app/models/ems_event.rb

```diff
@@ -164,6 +164,16 @@ def self.first_chained_event(ems_id, chain_id)
     EmsEvent.where(:ems_id => ems_id, :chain_id => chain_id).order(:id).first
   end
 
+  def self.parse_event_metadata(event) # event might be nil!
+    status_description = nil
+    if event.try(:event_type) == "datawarehouse_alert"
```

Comment:
Why would the method `parse_event_metadata` ever be called with a `nil` object?  The call `event.try(:class).try(:parse_event_metadata, event)` would prevent this from being calls.

Also, all the `try` logic in here is a code smell.  If this is really needed it would be better to return from the method early if the event is nil.

```ruby
def self.parse_event_metadata(event) # event might be nil!
  return [nil,nil,nil] if event.nil?
  ...
end  
```